web/pgadmin/browser/static/js/MainMenuFactory.js

```diff
@@ -110,7 +110,24 @@ export default class MainMenuFactory {
   }
 
   static getContextMenu(menuList) {
-    Menu.sortMenus(menuList);
-    return menuList;
+    let selectedNode=pgAdmin.Browser.tree.selected();
+    let d=pgAdmin.Browser.tree.itemData(selectedNode);
+    let selectedNodeFromNodes=pgAdmin.Browser.Nodes[d._type];
+    if (!_.isUndefined(selectedNodeFromNodes.showMenu)){
+      var showMenu = selectedNodeFromNodes.showMenu(d, selectedNode);
+      if (showMenu===false) {
+        return [new MenuItem({
```

Comment:
Merge the logic to create this menuItem for both Objects(browser.js  L: 478)menu and context menu

Reply:
Logic merged to a one to accommodate object menu and context menu.